src/dxgi/dxgi_output.cpp

```diff
@@ -46,8 +46,77 @@ namespace dxvk {
     const DXGI_MODE_DESC *pModeToMatch,
           DXGI_MODE_DESC *pClosestMatch,
           IUnknown       *pConcernedDevice) {
-    Logger::err("DxgiOutput::FindClosestMatchingMode: Not implemented");
-    return E_NOTIMPL;
+    if (pModeToMatch == nullptr) {
+      Logger::err("DxgiOutput::FindClosestMatchingMode: pModeToMatch is nullptr");
+      return DXGI_ERROR_INVALID_CALL;
+    }
+        
+    if (pClosestMatch == nullptr) {
+      Logger::err("DxgiOutput::FindClosestMatchingMode: pClosestMatch is nullptr");
+      return DXGI_ERROR_INVALID_CALL;
+    }
+
+    if (pModeToMatch->Format == DXGI_FORMAT_UNKNOWN && pConcernedDevice == nullptr) {
+      Logger::err("DxgiOutput::FindClosestMatchingMode: no pointer to device was provided for DXGI_FORMAT_UNKNOWN format");
+      return DXGI_ERROR_INVALID_CALL;
+    }
+
+    if (pModeToMatch->Format == DXGI_FORMAT_UNKNOWN) {
```

Comment:
What exactly is supposed to happen when `Format` is `UNKNOWN`? I can't find any documentation about it.

It's fine to leave this open for now, but since this is a legal thing to do, please add some kind of `// TODO implement` comment.

Reply:
There's some obfuscated stuff in MSDN:
https://msdn.microsoft.com/en-us/library/windows/desktop/bb174547(v=vs.85).aspx

>If `pConcernedDevice` is `NULL`, `Format` **CANNOT** be `DXGI_FORMAT_UNKNOWN`.

using some basic laws of logic, we can read this like

>If `pConcernedDevice` is NOT `NULL`, `Format` **COULD** be `DXGI_FORMAT_UNKNOWN`.

But `Format` in structures from `GetDisplayModeList()` cannot be `DXGI_FORMAT_UNKNOWN` by definition. 

There is way in case of `DXGI_FORMAT_UNKNOWN` and `pDevice != nullptr` we should perform additional format matching. It may be just ignoring of `Format` field or using some range of formats but MSDN nothing says about of criteria. :(